### PR TITLE
Adding r-fedmatch as a package

### PR DESCRIPTION
**Pull request by JNuss71**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

I used the [conda_r_skeleton_helper](https://github.com/bgruening/conda_r_skeleton_helper?tab=readme-ov-file) to expedite the package review process for adding the [fedmatch package](https://cran.r-project.org/web/packages/fedmatch).

### CODE REVIEW

**Comment by JNuss71** [comment]
@conda-forge/help-r, ready for review!

**Comment by JNuss71** [comment]
@conda-forge-admin, please ping team

File: recipes/r-fedmatch/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    - r-forcats
+    - r-magrittr
+    - r-purrr
+    - r-scales
+    - r-stringdist
+    - r-stringr
+  run:
+    - r-base
+    - {{ native }}gcc-libs         # [win]
+    - r-bh
```
Comment (mfansler) [inline recipes/r-fedmatch/meta.yaml:54]: Boost headers are only needed at build time. We usually remove it from `run:`, when only in `LinkingTo`, i.e.,

```suggestion
```

File: recipes/r-fedmatch/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  missing_dso_whitelist:
+    - '*/R.dll'  # [win]
+  number: 0
+  rpaths:
+    - lib/R/lib/
+    - lib/
+
+requirements:
+  build:
+    - {{ stdlib("c") }}
```
Comment (mfansler) [inline recipes/r-fedmatch/meta.yaml:26]: A little adjustment for win:

```suggestion
    - {{ stdlib("c") }}        # [not win]
    - {{ stdlib("m2w64_c") }}  # [win]
```

**Review by mfansler — CHANGES_REQUESTED** [review]
Thanks for adding this! ❤️ 

Mostly looks good - a couple minor changes, then should be good.

**Comment by mfansler** [comment]
R Team is happy to help maintain.

**Review by mfansler — APPROVED** [review]

**Pull request merged**